lib/experimental/modules/class-gutenberg-modules.php

```diff
@@ -0,0 +1,186 @@
+	 * @param string           $src               Full URL of the module, or path of the script relative to the WordPress root directory.
+	 * @param array            $dependencies      Optional. An array of module identifiers of the static and dynamic dependencies of this module. It can be an indexed array, in which case all the dependencies are static, or it can be an associative array, in which case it has to contain the keys `static` and `dynamic`.
+	 * @param string|bool|null $version           Optional. String specifying module version number. It is added to the URL as a query string for cache busting purposes. If SCRIPT_DEBUG is true, a timestamp is used. If it is set to false, a version number is automatically added equal to current installed WordPress version. If set to null, no version is added.
+	 */
+	public static function register( $module_identifier, $src, $dependencies = array(), $version = false ) {
+		// Register the module if it's not already registered.
+		if ( ! isset( self::$registered[ $module_identifier ] ) ) {
+			$deps = array(
+				'static'  => isset( $dependencies['static'] ) || isset( $dependencies['dynamic'] ) ? $dependencies['static'] ?? array() : $dependencies,
+				'dynamic' => isset( $dependencies['dynamic'] ) ? $dependencies['dynamic'] : array(),
+			);
+
+			self::$registered[ $module_identifier ] = array(
+				'src'          => $src,
+				'version'      => $version,
+				'dependencies' => $deps,
+			);
+		}
+	}
+
+	/**
+	 * Enqueues a module in the page.
+	 *
+	 * @param string $module_identifier The identifier of the module.
+	 */
+	public static function enqueue( $module_identifier ) {
+		// Add the module to the queue if it's not already there.
+		if ( ! in_array( $module_identifier, self::$enqueued, true ) ) {
+			self::$enqueued[] = $module_identifier;
+		}
+	}
+
+	/**
+	 * Returns the import map array.
+	 *
+	 * @return string The import map.
+	 */
+	public static function get_import_map() {
+		$imports = array();
+		foreach ( self::get_dependencies( self::$enqueued, array( 'static', 'dynamic' ) ) as $module_identifier => $module ) {
+			$imports[ $module_identifier ] = $module['src'] . self::get_version_query_string( $module['version'] );
+		}
+		return array( 'imports' => $imports );
+	}
+
+	/**
+	 * Prints the import map.
+	 */
+	public static function print_import_map() {
+		echo '<script type="importmap">' . wp_json_encode( self::get_import_map(), JSON_HEX_TAG | JSON_HEX_AMP ) . '</script>';
+	}
+
+	/**
+	 * Prints all the enqueued modules using <script type="module">.
+	 */
+	public static function print_enqueued_modules() {
+		foreach ( self::$enqueued as $module_identifier ) {
+			if ( isset( self::$registered[ $module_identifier ] ) ) {
+					$module = self::$registered[ $module_identifier ];
+					echo '<script type="module" src="' . $module['src'] . self::get_version_query_string( $module['version'] ) . '" id="' . $module_identifier . '"></script>';
```

Comment:
Should we add here a integrity check for array keys 'src' and 'version' ?

Reply:
There are `wp_print_script_tag` and `wp_get_script_tag` that are filterable and they provide basic sanitization for the values provided in attributes.

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/56143/commits/4e6c5da06fd85662b1c6eb120e382af455958200. Thanks folks!